fedimint-client/src/db.rs

```diff
@@ -441,9 +479,25 @@ pub async fn remove_old_and_persist_new_active_states(
 pub async fn remove_old_and_persist_new_inactive_states(
     dbtx: &mut DatabaseTransaction<'_>,
     new_inactive_states: Vec<DynState>,
+    module_instance_id: ModuleInstanceId,
 ) {
     // Remove all existing active states
-    dbtx.remove_by_prefix(&InactiveStateKeyPrefix::new()).await;
+    let states_to_remove = dbtx
```

Comment:
Don't you already have this one from the time you queried it?

Reply:
ah good point, fixed